File: backend/ModuMessenger/src/main/java/com/example/modumessenger/config/FeignConfig.java
package com.example.modumessenger.config;

import com.example.modumessenger.common.decoder.FeignErrorDecoder;
import feign.Logger;
import feign.Retryer;
import feign.codec.ErrorDecoder;
import org.springframework.context.annotation.Bean;

public class FeignConfig {

    @Bean
    Logger.Level feignLoggerLevel() {
        return Logger.Level.FULL;
    }

    @Bean
    public Retryer retryer() {
        return new Retryer.Default();
    }

    @Bean
    public ErrorDecoder errorDecoder() {
        return new FeignErrorDecoder();
    }
}

File: backend/chat-service/src/main/kotlin/com/example/chatservice/config/FeignConfig.kt
package com.example.chatservice.config

import feign.Logger
import feign.Retryer
import org.springframework.context.annotation.Bean

class FeignConfig {

    @Bean
    fun feignLoggerLevel(): Logger.Level = Logger.Level.FULL

    @Bean
    fun retryer(): Retryer = Retryer.Default()
}
